### DISCUSSION TITLE
onColumnSizingChange doesn't work by prop drilling

### DISCUSSION THREAD

**Discussion by joaoarieira** [TanStack/table#4473] [Q&A]
Hello, everyone.

If I pass columnSizing state and onColumnSizingChange as props to Table component, no column gets resized.
Why?

```
const Table = ({columnSizing, setColumnSizing}) => {
...
  const table = useReactTable({
      data,
      columns,
      getCoreRowModel: getCoreRowModel(),
      state: {
        columnSizing
      },
      onColumnSizingChange: setColumnSizing
    });
...
}
```

Example: https://codesandbox.io/s/gracious-gwen-ce63gb?file=/src/main.tsx:2807-3078

**Comment by planetmalone** [comment]
@joaoarieira Did you ever get this figured out? I'm running into a similar issue trying to update columnSizing from onStateChange when the Table is fully controlled.
